modules/compute_instance/main.tf

```diff
@@ -15,37 +15,39 @@
  */
 
 locals {
-  hostname      = "${var.hostname == "" ? "default" : var.hostname}"
-  num_instances = "${length(var.static_ips) == 0 ? var.num_instances : length(var.static_ips)}"
+  hostname      = var.hostname == "" ? "default" : var.hostname
+  num_instances = length(var.static_ips) == 0 ? var.num_instances : length(var.static_ips)
 
   # local.static_ips is the same as var.static_ips with a dummy element appended
   # at the end of the list to work around "list does not have any elements so cannot
   # determine type" error when var.static_ips is empty
-  static_ips = "${concat(var.static_ips, list("NOT_AN_IP"))}"
+  static_ips = concat(var.static_ips, ["NOT_AN_IP"])
 }
 
 ###############
 # Data Sources
 ###############
 
-data "google_compute_zones" "available" {}
+data "google_compute_zones" "available" {
+}
 
 #############
 # Instances
 #############
 
 resource "google_compute_instance_from_template" "compute_instance" {
-  provider = "google"
-  count    = "${local.num_instances}"
+  provider = google
+  count    = local.num_instances
   name     = "${local.hostname}-${format("%03d", count.index + 1)}"
-  zone     = "${data.google_compute_zones.available.names[count.index % length(data.google_compute_zones.available.names)]}"
+  zone     = data.google_compute_zones.available.names[count.index % length(data.google_compute_zones.available.names)]
 
   network_interface {
-    network            = "${var.network}"
-    subnetwork         = "${var.subnetwork}"
-    subnetwork_project = "${var.subnetwork_project}"
-    network_ip         = "${length(var.static_ips) == 0 ? "" : element(local.static_ips, count.index)}"
+    network            = var.network
+    subnetwork         = var.subnetwork
+    subnetwork_project = var.subnetwork_project
+    network_ip         = length(var.static_ips) == 0 ? "" : element(local.static_ips, count.index)
```

Comment:
Please check if it is better to use "null"

Reply:
I'll check